openxr/examples/opengl_xlib.rs

```diff
@@ -0,0 +1,275 @@
+#[macro_use]
+extern crate glium;
+
+use std::ptr;
+use openxr as xr;
+use x11::{xlib, glx};
+use std::ffi::{c_void, CString};
+use glium::texture::{DepthTexture2d, DepthFormat, UncompressedFloatFormat, MipmapsOption};
+
+const GL_RGBA8: u32 = 0x8058;
+
+pub struct Backend {
+    context: glx::GLXContext,
+    display: *mut xlib::Display,
+    visual: *mut xlib::XVisualInfo,
+    fb_config: glx::GLXFBConfig,
+    drawable: x11::xlib::Drawable,
+}
+
+impl Backend {
+    pub fn new() -> Self {
+        let mut fbcount = 0;
+        let visual_attribs = [0];
+
+        let context_attribs = [
+            glx::GLX_RGBA,
+            glx::GLX_DEPTH_SIZE,
+            24,
+            glx::GLX_DOUBLEBUFFER,
+            0,
+        ];
+        unsafe {
+            let display = xlib::XOpenDisplay(ptr::null());
+            let root = xlib::XDefaultRootWindow(display);
+            let visual = glx::glXChooseVisual(display, 0, context_attribs.as_ptr() as *mut _);
+            let context =
+                glx::glXCreateContext(display, visual, 0 as glx::GLXContext, true as xlib::Bool);
+            let fb_config = glx::glXChooseFBConfig(
+                display,
+                xlib::XDefaultScreen(display),
+                visual_attribs.as_ptr(),
+                &mut fbcount,
+            );
+
+            glx::glXMakeCurrent(display, root, context);
+
+            Self {
+                context,
+                display,
+                visual,
+                fb_config: *fb_config,
+                drawable: root,
+            }
+        }
+    }
+    pub unsafe fn xr_session_create_info(&self) -> xr::opengl::SessionCreateInfo {
+        let visualid = { *self.visual }.visualid as u32;
+        xr::opengl::SessionCreateInfo::Xlib {
+            x_display: self.display as *mut _,
+            glx_fb_config: self.fb_config as *mut _,
+            glx_drawable: self.drawable,
+            visualid: visualid,
+            glx_context: self.context as *mut _,
+        }
+    }
+}
+
+impl Drop for Backend {
+    fn drop(&mut self) {
+        unsafe {
+            x11::xlib::XFree(self.fb_config as *mut _);
+            x11::xlib::XCloseDisplay(self.display);
```

Comment:
This should be the last call, just in case the other calls reference the display internally somehow.